hw_2/.ipynb_checkpoints/hw_2-checkpoint.ipynb

```diff
@@ -0,0 +1,363 @@
+    "def try_ex_func():\n",
+    "    user_input = input(\"Введите 4 отношения частот, фигуры Лиссажу для которых Вы хотите получить (например 1:1, 2:1, 4:3, 5:7): \")\n",
+    "    try:\n",
+    "        if (user_input.find(':') == -1):\n",
+    "            print(\"Неверный ввод.\")\n",
+    "            return try_ex_func()\n",
+    "\n",
+    "        frequencies = [(int(el.split(':')[0]), int(el.split(':')[1])) for el in user_input.strip().split(', ')]   #Питон могет)\n",
+    "    except Exception:\n",
+    "        print(\"Неверный ввод.\")\n",
+    "        return try_ex_func()\n",
+    "        \n",
+    "    else:\n",
+    "        print(\"Ниже приведены 4 фигуры Лиссажу.\")\n",
+    "        return frequencies\n",
+    "\n",
+    "freq = try_ex_func()\n",
+    "\n",
+    "\n",
+    "fig, ax = plt.subplots(2, 2)\n",
+    "\n",
+    "A = np.linspace(0, 6*np.pi, 1000)\n",
+    "\n",
+    "ax[0, 0].plot(np.cos(freq[0][0]*A), np.sin(freq[0][1]*A), color = 'black')\n",
+    "ax[0, 0].scatter(np.cos(freq[0][0]*A), np.sin(freq[0][1]*A), color = 'red', s = 5)\n",
+    "\n",
+    "ax[1, 0].plot(np.cos(freq[1][0]*A), np.sin(freq[1][1]*A), color = 'black')\n",
+    "ax[1, 0].scatter(np.cos(freq[1][0]*A), np.sin(freq[1][1]*A), color = 'red', s = 5)\n",
+    "\n",
+    "ax[0, 1].plot(np.cos(freq[2][0]*A), np.sin(freq[2][1]*A), color = 'black')\n",
+    "ax[0, 1].scatter(np.cos(freq[2][0]*A), np.sin(freq[2][1]*A), color = 'red', s = 5)\n",
+    "\n",
+    "ax[1, 1].plot(np.cos(freq[3][0]*A), np.sin(freq[3][1]*A), color = 'black')\n",
+    "ax[1, 1].scatter(np.cos(freq[3][0]*A), np.sin(freq[3][1]*A), color = 'red', s = 5)\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "a = np.array([np.cos(freq[3][0]*A), np.sin(freq[3][1]*A)])\n",
+    "a_squared = np.multiply(a,a)\n",
+    " \n",
+    "dist = a_squared[0] + a_squared[1] #удаленность каждой точки от центра координат в квадрате\n",
+    "\n",
+    "plt.title('Гистограма для последнего введенного соотношения частот')\n",
+    "plt.hist(dist, 30, density=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "b = 2*np.eye(10, dtype=int)\n",
+    "\n",
+    "b -= np.diag(np.ones(9, dtype=int), 1)\n",
+    "b -= np.diag(np.ones(9, dtype=int), -1)\n",
+    "\n",
+    "print(b)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "X = np.linspace(0, 10, 10) + np.random.normal(0,1,10) #добавил шумов\n",
+    "Y = np.linspace(0, 10, 10) + np.random.normal(0,1,10) #добавил шумов\n",
+    "\n",
+    "A = np.vstack([X, np.ones(len(X))]).T\n",
```

Comment:
It would be better to use np.ones(X.shape) to make sure you get X-conformant array